### ISSUE TITLE
Some signup and reportback_submission documents have a null user

### ISSUE THREAD

**Issue by aaronschachter** [DoSomething/gambit#827]
_BUG_
-------------------
#### Current Behavior
Refs https://github.com/DoSomething/gambit/pull/826#issuecomment-288544291

* A small number of `signup` documents (48) have a null `user` in production.
    * db.getCollection('signups').find( { user : { $type: 10 } } )
* When testing signups with a new user (and fake mobile number), the draft `reportback_submission` is stored with a null user.
    * There are only 6 `reportback_submissions` documents with a null user on production

#### Desired Behavior
The `user` property should store the Signup or Reportback Submission User's Northstar ID.

#### Steps to Replicate
Don't know yet

#### Why This Matters
For Signups with a null user -- I'm not entirely sure it does, but want to document this somewhere. I think this happens when Quicksilver posts external Signups to the Gambit `/signups` endpoint, and the GET request to find the Northstar user fails (e.g. https://dosomething.slack.com/archives/C1V0M6RPE/p1490230443751810). I've been digging into Signups and Users here: https://gist.github.com/aaronschachter/9725c8d885808361f6751d40ae39e473 and so far a common thread seems to be they are external signups.

For Reportback Submissions with a null user -- let's pinpoint why this is happening.

**Comment by aaronschachter** [comment]
Small number of `signup:0` that show up in the logs, which occur when can replicate this bug.
```
» 21 Mar 2017 12:18:16.229 125 <190>1 2017-03-21T19:18:16.181248+00:00 app web.1 - - [32minfo[39m: loaded signup:0 Context
» 21 Mar 2017 12:49:29.823 138 <190>1 2017-03-21T19:49:29.578156+00:00 app web.1 - - [32minfo[39m: Signup.post created signup:0 Context
» 21 Mar 2017 12:49:29.823 125 <190>1 2017-03-21T19:49:29.582152+00:00 app web.1 - - [32minfo[39m: loaded signup:0 Context
» 21 Mar 2017 12:55:17.566 138 <190>1 2017-03-21T19:55:17.494226+00:00 app web.1 - - [32minfo[39m: Signup.post created signup:0 Context
» 21 Mar 2017 12:55:17.566 125 <190>1 2017-03-21T19:55:17.504371+00:00 app web.1 - - [32minfo[39m: loaded signup:0 Context
» 21 Mar 2017 13:16:24.855 138 <190>1 2017-03-21T20:16:24.737115+00:00 app web.1 - - [32minfo[39m: Signup.post created signup:0 Context
» 21 Mar 2017 13:16:24.855 125 <190>1 2017-03-21T20:16:24.741074+00:00 app web.1 - - [32minfo[39m: loaded signup:0 Context
» 21 Mar 2017 13:55:07.543 138 <190>1 2017-03-21T20:55:07.317044+00:00 app web.1 - - [32minfo[39m: Signup.post created signup:0 Context
» 21 Mar 2017 13:55:07.543 125 <190>1 2017-03-21T20:55:07.321584+00:00 app web.1 - - [32minfo[39m: loaded signup:0 Context
```

**Comment by aaronschachter** [comment]
I'm down the rabbit hole a little bit, but one instance where this occur might be if someone responds YES to a broadcast twice -- Gambit makes a POST request to Signup for a 2nd time because it's not yet finished saving the User's info for the 1st "yes".

I found this for https://secure.mcommons.com/profiles/203718616

<img width="1258" alt="screen shot 2017-03-24 at 10 37 23 am" src="https://cloud.githubusercontent.com/assets/1236811/24306547/30cce802-107e-11e7-9c18-d57d88780df6.png">

**Comment by aaronschachter** [comment]
I'm getting somewhere. Locally-- I created a new user with a dummy number, 201703241023 and signed up for 2900. New user has NS ID 58d5563d10707d4d987b75b3

mobile: 201703241057
user:58d55e0910707d4d987b75b5

A signup is created: 2844891
Next request will lookup UID 58d5563d10707d4d987b75b3 and Campaign 2900:
```
10:59:13 AM web.1 |  debug: Signup.post(58d55e0910707d4d987b75b5, 2900, undefined)
10:59:14 AM web.1 |  debug: stathat: gambit-dev - phoenix: POST signups 200
10:59:14 AM web.1 |  info: Signup.post created signup:0
10:59:14 AM web.1 |  info: loaded signup:0
```
https://northstar-thor.dosomething.org/v1/signups?user=58d55e0910707d4d987b75b5&campaigns=2900
results in empty set:
<img width="629" alt="screen shot 2017-03-24 at 11 01 15 am" src="https://cloud.githubusercontent.com/assets/1236811/24307397/3cf7dae4-1081-11e7-97e2-9e306c0c693e.png">

**Comment by aaronschachter** [comment]
There are no signups for my user: https://northstar-thor.dosomething.org/v1/signups?user=5547be89469c64ec7d8b518d

**Comment by weerd** [comment]
@aaronschachter if you hit Thor directly instead of proxying through Northstar you _do_ see the signups: https://thor.dosomething.org/api/v1/signups?users=5547be89469c64ec7d8b518d

**Comment by aaronschachter** [comment]
We could potentially make the change in Gambit to query Phoenix Signups instead of Northstar, but making the change in our mobile apps would be a much bigger lift.

This Gambit issue is caused by two external issues:
* https://github.com/DoSomething/phoenix/issues/7329 per above
* https://github.com/DoSomething/phoenix-js/issues/17 -- The Phoenix JS library is not correctly interpreting a failed Signup POST

**Comment by aaronschachter** [comment]
https://github.com/DoSomething/phoenix-js/pull/18 has been merged. On our next deploy, Signups that fail will trigger an mData error, which would have helped flag breaking changes from https://github.com/DoSomething/phoenix/pull/7328.

**Comment by weerd** [comment]
Hmmm, it's definitely odd that the signup `user.id` would be `null` though.

**Comment by aaronschachter** [comment]
It's breaking more things than expected in Gambit... so making the switch just to query Phoenix instead of NS for signups doesn't fix this bug yet. It's happening for all my SIgnups on https://www.dosomething.org/api/v1/signups?user=5547be89469c64ec7d8b518d

**Comment by weerd** [comment]
I think I may have found something... when it retrieves the user data for the signup (after already having used the converted NS ID to get the signup), it doesn't convert the NS ID to grab the user to add to the signup object.

**Comment by aaronschachter** [comment]
Closed by #830 and https://github.com/DoSomething/phoenix/pull/7331
